### PR TITLE
Rebase of @bprashant Flannel out of the box

### PR DESCRIPTION
**Pull request by gmarek**
cc @bprashanth @thockin @davidopp

### CODE REVIEW

File: cluster/saltbase/salt/kubelet/default [hunk 1]
```diff
@@ -117,4 +122,4 @@
 {% endif -%}
 
 # test_args has to be kept at the end, so they'll overwrite any prior configuration
-DAEMON_ARGS="{{daemon_args}} {{api_servers_with_port}} {{debugging_handlers}} {{hostname_override}} {{cloud_provider}} {{config}} {{manifest_url}} --allow-privileged={{pillar['allow_privileged']}} {{pillar['log_level']}} {{cluster_dns}} {{cluster_domain}} {{docker_root}} {{kubelet_root}} {{configure_cbr0}} {{cgroup_root}} {{system_container}} {{pod_cidr}} {{ master_kubelet_args }} {{cpu_cfs_quota}} {{network_plugin}} {{test_args}}"
+DAEMON_ARGS="{{daemon_args}} {{api_servers_with_port}} {{debugging_handlers}} {{hostname_override}} {{cloud_provider}} {{config}} {{manifest_url}} --allow-privileged={{pillar['allow_privileged']}} {{pillar['log_level']}} {{cluster_dns}} {{cluster_domain}} {{docker_root}} {{kubelet_root}} {{configure_cbr0}} {{cgroup_root}} {{system_container}} {{pod_cidr}} {{ master_kubelet_args }} {{cpu_cfs_quota}} {{network_plugin}} {{experimental_flannel_overlay}} {{test_args}}"
```
Comment (wojtek-t) [inline cluster/saltbase/salt/kubelet/default]: Note: this place is different the the original PR - but it is trivial rebase.

Comment (gmarek) [inline cluster/saltbase/salt/kubelet/default]: + a fix:) {{test_args}} has to be always last, which wasn't true in the original PR.

**Comment by wojtek-t** [comment]
The rebase was pretty trivial - LGTM

**Comment by gmarek** [comment]
@k8s-bot test this

**Comment by eparis** [comment]
@gmarek you can use `cla: human-approved` for cases like this the bot doesn't understand.  The bot will still add `cla: no` next time someone pushes code, but the submit queue uses human-approved as an override

**Comment by gmarek** [comment]
@eparis - thanks, I didn't know.

I'm merging this PR by hand, as it's important for us and submit queue is blocked by the test flakynes for the (nearly) whole day today.

**Pull request merged**